rbac/management/utils.py

```diff
@@ -280,6 +280,12 @@ def validate_group_name(name):
         message = f"{name} is reserved, please use another name."
         raise serializers.ValidationError({key: _(message)})
 
+def check_group_exists(groupName):
+    groupTenant = Group.objects.get(name=groupName)
+    if Group.objects.filter(name=groupName).exists():
```

Comment:
We need to include the tenant in this query; otherwise, it will return results for other tenants.

Alternatively, we could execute a single query that returns the group object (or None), and then you can extract the tenant object from the group object and use org_id in error message. (org_id is known for user, tenant_id is rather internal RBAC thing )

Reply:
You might want to instead catch a unique constraint violation when the create happens. This does a few things for you:

1. It fixes the query (would only error for the same tenant)
2. It catches race conditions (this check would not catch if a group was concurrently added, you will still get the constraint violation error)
3. It eliminates a second query (only does the one insert, rather than read + insert)

But, up to y'all :) 🤘🏻